webapp/src/js/services/db-sync.js

```diff
@@ -73,20 +75,34 @@ angular
             return;
           }
           const remote = DB({ remote: true });
+          const onDeniedPromises = [];
           return DB()
             .replicate[direction.name](remote, direction.options)
             .on('denied', function(err) {
-              // In theory this could be caused by 401s
-              // TODO: work out what `err` looks like and navigate to login
-              // when we detect it's a 401
               $log.error(`Denied replicating ${direction.name} remote server`, err);
+              if (direction.onDenied) {
+                onDeniedPromises.push(direction.onDenied(err));
+              }
             })
             .on('error', function(err) {
               $log.error(`Error replicating ${direction.name} remote server`, err);
             })
             .then(info => {
-              $log.debug(`Replication ${direction.name} successful`, info);
-              return;
+              // no failures to update
+              if (!onDeniedPromises.length) {
+                $log.debug(`Replication ${direction.name} successful`, info);
+                return;
+              }
+
+              // if any of the denied docs were touched, restart replication so they get re-uploaded
+              return $q.all(onDeniedPromises).then(results => {
+                if (results.some(result => result)) {
+                  return replicate(direction);
```

Comment:
I'm tempted to not immediately try to replicate again to reduce complexity. The touched rev would be replicated up in the next batch. We also have this issue scheduled for 3.9.0 which would trigger a fresh replication immediately: https://github.com/medic/cht-core/issues/5738

Reply:
Sure, it's the way I thought about implementing it at 1st as well. 
Then I saw that we report in the hamburger menu that "All reports are synced" regardless, which would be false when we touch these docs. 
In 5738 we're going to have to take this race-condition (that docs might get edited while replicating up) into account, and we'd "queue" the next replication somehow.